packages/app/contexts/TokenListContext.tsx

```diff
@@ -97,9 +105,54 @@ export const TokenListProvider = ({ children }: PropsWithChildren) => {
     setupTokenList();
   }, [address, setupTokenList]);
 
+  /**
+   * Once the tokenList is defined and we have an address
+   * connected, we batch fetch all the ERC-20 token balances
+   * for the connected wallet and the supported token list,
+   * then we desc. sort them
+   */
+  useEffect(() => {
+    if (address) {
+      const fetchErc20Balances = async () => {
+        try {
+          const batchFetchdata = await multicall({
+            contracts: tokenList.map((token) => ({
+              address: token.address as `0x${string}`,
+              abi: erc20ABI,
+              functionName: "balanceOf",
+              args: [address as `0x${string}`],
+              chainId: token.chainId,
+            })),
+            allowFailure: true,
+          });
+
+          const listWithBalances = tokenList
+            .map((token, index) => ({
+              ...token,
+              balance: formatUnits(
+                batchFetchdata[index]?.result as bigint,
```

Comment:
Just wondering if it's safe to assume the index will match the correct `token` on `batchFetchdata` 🤔

Reply:
It is an array based on the `tokenList`, it _should_ have the same indexing